### PR TITLE
Rename gRPC-specific symbols

### PR DESCRIPTION
**Pull request by tiziano88**
- Rename `OakExports` to `OakGrpcEntrypoint`
- Rename `OakNode` to `OakGrpcServerNode`

The previous versions suggested that they are the canonical way of
implementing Oak nodes, while in fact they are specific to gRPC server
nodes.

### Checklist

- [ ] Pull request affects core Oak functionality (e.g. runtime, SDK, ABI)
  - [ ] I have written tests that cover the code changes.
  - [ ] I have checked that these tests are run by [Cloudbuild](cloudbuild.yaml)
  - [x] I have updated [documentation](docs/) accordingly.
  - [ ] I have raised an [issue](https://github.com/project-oak/oak/issues) to
        cover any TODOs and/or unfinished work.
- [ ] Pull request includes prototype/experimental work that is under
      construction.

### CODE REVIEW

File: docs/sdk.md [hunk 1]
```diff
@@ -70,12 +70,12 @@ world over gRPC, via the implicit [gRPC pseudo-Node](concepts.md#pseudo-nodes)
 that has a channel to the Application's initial Node.
 
 An Oak Node that interacts with gRPC typically implements the
-[`oak::grpc::OakNode`](https://project-oak.github.io/oak/sdk/oak/grpc/trait.OakNode.html)
+[`oak::grpc::OakGrpcServerNode`](https://project-oak.github.io/oak/sdk/oak/grpc/trait.OakGrpcServerNode.html)
```
Comment (daviddrysdale) [inline docs/sdk.md]: I don't think this rename is necessary: the module name already indicates that `oak::grpc::OakNode` is gRPC-specific.

Comment (tiziano88) [reply docs/sdk.md]: I wanted to emphasize that it is a server though. How about `ServerNode` (omitting the `OakGrpc` prefix)?

Comment (daviddrysdale) [reply docs/sdk.md]: SGTM

Comment (tiziano88) [reply docs/sdk.md]: How about `derive(OakExports)` -> `derive(GrpcServerEntrypoint)`?

**Comment by tiziano88** [comment]
Closing in favour of #555.

**Pull request closed without merging**